File: 15-07-2025_Valid_Word.cpp
#include <string>
#include <cctype>
using namespace std;

class Solution {
public:
    bool isValid(string word) {
        if (word.length() < 3) return false;

        for (char c : word) {
            if (!isalnum(c)) return false;
        }

        string lower;
        for (char c : word) {
            lower += tolower(c);
        }

        bool hasVowel = false, hasConsonant = false;

        for (char c : lower) {
            if (string("aeiou").find(c) != string::npos) {
                hasVowel = true;
            } else if (string("bcdfghjklmnpqrstvwxyz").find(c) != string::npos) {
                hasConsonant = true;
            }
        }

        return hasVowel && hasConsonant;
    }
};

File: 15-07-2025_Valid_Word.java
public class Solution {
    public boolean isValid(String word) {
        if (word.length() < 3 || !word.matches("[a-zA-Z0-9]+")) {
            return false;
        }

        String lower = word.toLowerCase();
        boolean hasVowel = false, hasConsonant = false;

        for (char c : lower.toCharArray()) {
            if ("aeiou".indexOf(c) >= 0) {
                hasVowel = true;
            } else if ("bcdfghjklmnpqrstvwxyz".indexOf(c) >= 0) {
                hasConsonant = true;
            }
        }

        return hasVowel && hasConsonant;
    }
}
